policy-controller/k8s/index/src/pod.rs

```diff
@@ -68,6 +69,76 @@ pub(crate) fn tcp_port_names(spec: Option<k8s::PodSpec>) -> HashMap<String, Port
     port_names
 }
 
+/// Gets the container probe ports for a Pod.
+///
+/// The result is a mapping for each probe port exposed by a container in the
+/// Pod and the paths for which probes are expected. When a TCP probe is
+/// configured there is no path, so we use '/'.
+pub(crate) fn get_container_probes(
+    spec: &Option<k8s::PodSpec>,
+    port_names: &HashMap<String, PortSet>,
+) -> PortMap<HashSet<String>> {
+    let mut probes = PortMap::<HashSet<String>>::default();
+    if let Some(spec) = spec {
+        for container in spec.containers.clone().into_iter() {
+            container
+                .liveness_probe
+                .map(|probe| set_probe_ports(probe, &mut probes, port_names));
+            container
+                .readiness_probe
+                .map(|probe| set_probe_ports(probe, &mut probes, port_names));
+            container
+                .startup_probe
+                .map(|probe| set_probe_ports(probe, &mut probes, port_names));
+        }
+    }
+    return probes;
+}
+
+fn set_probe_ports(
+    probe: corev1::Probe,
+    probes: &mut PortMap<HashSet<String>>,
+    port_names: &HashMap<String, PortSet>,
+) {
+    if let Some(http) = probe.http_get {
+        if let Some(path) = http.path {
+            match http.port {
+                IntOrString::Int(port) => {
+                    if let Ok(p) = u16::try_from(port).and_then(NonZeroU16::try_from) {
+                        let paths = probes.entry(p).or_insert(Default::default());
+                        paths.insert(path);
+                    }
+                }
+                IntOrString::String(port) => {
+                    if let Some(named) = port_names.get(&port) {
+                        for p in named.into_iter() {
+                            let paths = probes.entry(*p).or_insert(Default::default());
```

Comment:
nit: clippy suggests `or_insert_with(Default::default)`...but I think if you make that change, it'll just say to do this:
```suggestion
                            let paths = probes.entry(*p).or_default();
```